src/App.tsx

```diff
@@ -1,10 +1,51 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
+import { getTodos } from './api/todos';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { Todo } from './types/Todo';
 import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './components/ErrorMessage';
+import { ApiError } from './types/ApiError';
+import { filterTodos, FilterType } from './helpers/filterTodos';
 
-const USER_ID = 0;
+const USER_ID = 6950;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(false);
+  const [errorType, setErrorType] = useState(ApiError.None);
```

Comment:
as for me, you can use only 1 state to handle error like
```
const [error, setError] = useState(ApiError.None)
```

then pass your error to `<ErrorMessage />` component, and check 
```
if (error === ApiError.None) {
  return null;
}
```